java/client/src/org/openqa/selenium/support/PageFactory.java

```diff
@@ -122,19 +126,25 @@ private static void proxyFields(FieldDecorator decorator, Object page, Class<?>
     }
   }
 
-  private static <T> T instantiatePage(WebDriver driver, Class<T> pageClassToProxy) {
+  @SuppressWarnings("unchecked")
+  private static <T> T instantiatePage(SearchContext context, Class<T> pageClassToProxy) {
     try {
-      try {
-        Constructor<T> constructor = pageClassToProxy.getConstructor(WebDriver.class);
-        return constructor.newInstance(driver);
-      } catch (NoSuchMethodException e) {
-        return pageClassToProxy.newInstance();
+      Constructor<?>[] availableConstructors = pageClassToProxy.getDeclaredConstructors();
+      for (Constructor<?> c: availableConstructors){
+
+          Class<?>[] parameterTypes = c.getParameterTypes();
+          if (parameterTypes.length != 1)
+            continue;
```

Comment:
it is a good practice to always enclose condition block into curly brackets, even if there is only one operator